src/form/en_ivr.jl

```diff
@@ -0,0 +1,1763 @@
+		c_rating::Vector{<:Real};
+		report::Bool=true
+	)
+
+For IVR models with explicit neutrals,
+imposes a bound on the current magnitude per conductor 
+at both ends of the branch (total current, i.e. including shunt contributions).
+
+```
+cr_fr^2 + ci_fr^2 <= c_rating^2
+cr_to^2 + ci_to^2 <= c_rating^2
+```
+"""
+function constraint_mc_branch_current_limit(pm::AbstractExplicitNeutralIVRModel, nw::Int, f_idx::Tuple{Int,Int,Int}, t_idx::Tuple{Int,Int,Int}, f_connections::Vector, t_connections::Vector, c_rating::Vector{<:Real}; report::Bool=true)
+    cr_fr = var(pm, nw, :cr, f_idx)
+    ci_fr = var(pm, nw, :ci, f_idx)
+    cr_to = var(pm, nw, :cr, t_idx)
+    ci_to = var(pm, nw, :ci, t_idx)
+
+    cnds_finite_rating = [c for (c,r) in enumerate(c_rating) if r<Inf]
+    JuMP.@constraint(pm.model, [c in cnds_finite_rating], cr_fr[c]^2+ci_fr[c]^2 <= c_rating[c]^2)
+    JuMP.@constraint(pm.model, [c in cnds_finite_rating], cr_to[c]^2+ci_to[c]^2 <= c_rating[c]^2)
+end
+
+
+"""
+	function constraint_mc_thermal_limit_from(
+		pm::AbstractExplicitNeutralIVRModel,
+		nw::Int,
+		f_idx::Tuple{Int,Int,Int},
+		f_connections::Vector{Int},
+		rate_a::Vector{<:Real}
+	)
+
+For IVR models with explicit neutrals,
+imposes a bound on the from-side line power magnitude.
+"""
+function constraint_mc_thermal_limit_from(pm::AbstractExplicitNeutralIVRModel, nw::Int, f_idx::Tuple{Int,Int,Int}, f_connections::Vector{Int}, rate_a::Vector{<:Real})
+    vr_fr = [var(pm, nw, :vr, f_idx[1])[t] for t in f_connections]
+    vi_fr = [var(pm, nw, :vi, f_idx[1])[t] for t in f_connections]
+    cr_fr = var(pm, nw, :cr, f_idx)
+    ci_fr = var(pm, nw, :ci, f_idx)
+
+    for idx in 1:length(rate_a)
+        if rate_a[idx]<Inf
+            psw_fr_idx = JuMP.@NLexpression(pm.model,  vr_fr[idx]*cr_fr[idx] + vi_fr[idx]*ci_fr[idx])
+            qsw_fr_idx = JuMP.@NLexpression(pm.model, -vr_fr[idx]*ci_fr[idx] + vi_fr[idx]*cr_fr[idx])
+
+            JuMP.@NLconstraint(pm.model, psw_fr_idx^2 + qsw_fr_idx^2 <= rate_a[idx]^2)
+        end
+    end
+end
+
+
+"""
+	function constraint_mc_thermal_limit_to(
+		pm::AbstractExplicitNeutralIVRModel,
+		nw::Int,
+		t_idx::Tuple{Int,Int,Int},
+		t_connections::Vector{Int},
+		rate_a::Vector{<:Real}
+	)
+
+For IVR models with explicit neutrals,
+imposes a bound on the to-side line power magnitude.
+"""
+function constraint_mc_thermal_limit_to(pm::AbstractExplicitNeutralIVRModel, nw::Int, t_idx::Tuple{Int,Int,Int}, t_connections::Vector{Int}, rate_a::Vector{<:Real})
+    vr_to = [var(pm, nw, :vr, t_idx[1])[t] for t in t_connections]
+    vi_to = [var(pm, nw, :vi, t_idx[1])[t] for t in t_connections]
+    cr_to = var(pm, nw, :cr, t_idx)
+    ci_to = var(pm, nw, :ci, t_idx)
+
+    for idx in 1:length(rate_a)
+        if rate_a[idx]<Inf
+            psw_to_idx = JuMP.@NLexpression(pm.model,  vr_to[idx]*cr_to[idx] + vi_to[idx]*ci_to[idx])
+            qsw_to_idx = JuMP.@NLexpression(pm.model, -vr_to[idx]*ci_to[idx] + vi_to[idx]*cr_to[idx])
```

Comment:
It looks like you wrote a generic `thermal_limit_to` function, so the `psw_to` and `qsw_to` make it a little confusing, I think change to just pt and qt

also, update the docstring to indicate that the constraint is generic to edge types

Reply:
This constraint is intended for branches, and not for a generic edge type. I did not include `branch` in the constraint name to be in line with previous naming conventions. Should I follow the existing naming scheme or correct it here to include the branch qualifier?